### PR TITLE
[storage] Fixed storage crash.

### PR DESCRIPTION
**Pull request by ygorshenin**
@milchakov @bykoianko PTAL

### CODE REVIEW

File: storage/storage_tests/storage_tests.cpp [hunk 1]
```diff
@@ -1789,5 +1795,31 @@ UNIT_TEST(StorageTest_GetTopmostNodesForWithLevel)
...
+  storage.Init(&OnCountryDownloaded /* didDownload */,
+               [](TCountryId const &, TLocalFilePtr const) { return false; } /* willDelete */);
+  storage.SetDownloaderForTesting(make_unique<TestMapFilesDownloader>());
+  storage.SetCurrentDataVersionForTesting(1234);
+
+  auto const countryId = storage.FindCountryIdByFile("Uruguay");
+  auto const countryFile = storage.GetCountryFile(countryId);
 
+  // To prevent interference from other tests and on other tests it's
+  // better to remove temprorary downloader files.
```
Comment (mpimenov) [inline storage/storage_tests/storage_tests.cpp]: s/temprorary/temporary/

File: storage/storage_tests/storage_tests.cpp [hunk 2]
```diff
@@ -1789,5 +1795,31 @@ UNIT_TEST(StorageTest_GetTopmostNodesForWithLevel)
...
+
+  storage.Init(&OnCountryDownloaded /* didDownload */,
+               [](TCountryId const &, TLocalFilePtr const) { return false; } /* willDelete */);
+  storage.SetDownloaderForTesting(make_unique<TestMapFilesDownloader>());
+  storage.SetCurrentDataVersionForTesting(1234);
+
+  auto const countryId = storage.FindCountryIdByFile("Uruguay");
+  auto const countryFile = storage.GetCountryFile(countryId);
 
+  // To prevent interference from other tests and on other tests it's
```
Comment (mpimenov) [inline storage/storage_tests/storage_tests.cpp]: interference with other tests?
influence on and from other tests?

File: storage/storage.hpp [hunk 3]
```diff
@@ -648,7 +648,8 @@ class Storage : public diffs::Manager::Observer
                                                   MapFilesDownloader::TProgress const & downloadingMwmProgress,
                                                   TCountriesSet const & mwmsInQueue) const;
 
-  void CorrectJustDownloadedAndQueue(TQueue::iterator justDownloadedItem);
+  void PushToJustDownloaded(TQueue::iterator justDownloadedItem);
+  void PeekFromQueue(TQueue::iterator it);
```
[the remark this answers is not in the document]
Comment (ygorshenin) [reply storage/storage.hpp]: Pop sounds good.

File: storage/storage_tests/storage_tests.cpp (hunk 1 above)
Comment (ygorshenin) [reply storage/storage_tests/storage_tests.cpp]: Done

File: storage/storage_tests/storage_tests.cpp (hunk 2 above)
Comment (ygorshenin) [reply storage/storage_tests/storage_tests.cpp]: "interference" SGTM

**Comment by ygorshenin** [comment]
@mpimenov @bykoianko PTAL

**Comment by mpimenov** [comment]
@ygorshenin Have you pushed the changes?

**Comment by ygorshenin** [comment]
Please, take a look at https://github.com/mapsme/omim/pull/7410. We decided to put this on release-76.

**Pull request closed without merging**